test/rule_definition_tools.jl

```diff
@@ -183,6 +187,25 @@ end
             end
         end
 
+        @testset "Functors" begin
+            (f::NonDiffExample)(y) = fill(7.5, 100)[f.x + y]
+            @non_differentiable (::NonDiffExample)(::Any)
+            @test frule((Composite{NonDiffExample}(x=1.2), 2.3), NonDiffExample(3), 2) ==
+                (7.5, DoesNotExist())
+            res, pullback = rrule(NonDiffExample(3), 2)
+            @test res == 7.5
+            @test pullback(4.5) == (DoesNotExist(), DoesNotExist())
```

Comment:
Should we make this `NO_FIELDS` as well?
```suggestion
            @test pullback(4.5) == (NO_FIELDS, DoesNotExist())
```

Reply:
See the discussion above, I have now changed it to just always return `DoesNotExist()`.

Reply:
Ohh I now understand that discussion, I misread it initially in a confusing way...

Also, in my head `NO_FIELDS` was `DoesNotExist()` not `Zero()` so that was also confusing 😂